### PR TITLE
feat: check for new versions on launch

### PR DESCRIPTION
**Pull request by juliensimon**
## Summary
- Adds a rate-limited GitHub Releases check (once per 24h) so DMG users learn when a newer Canopy is available — Homebrew users get this for free, but DMG users were previously stranded on whatever they first downloaded.
- New `UpdateChecker` service with a pure, numeric semver comparator and a typed `LatestRelease` decoder.
- Update status is surfaced in **About** (status row + Download button + manual "Check Now") and **Settings** (new "Updates" section with toggle and manual check).
- When a newer version is found and the app is in the background, posts a macOS notification via the existing osascript path (extracted into a small reusable helper).
- 9 new unit tests cover semver edge cases (numeric vs lexical, leading `v`, prerelease suffix, missing components) and JSON decoding of the GitHub Releases payload.

## Implementation notes
- Rate-limit timestamp lives in `UserDefaults` (`canopy.lastUpdateCheck`), not `settings.json` — it's ephemeral cache state, not user preference.
- `compareSemver` deliberately drops prerelease suffixes and treats missing components as zero — Canopy's versioning is simple `MAJOR.MINOR.PATCH`.
- Sets `User-Agent: Canopy/<version>` because GitHub rejects unauthenticated requests with no UA.
- `/releases/latest` already excludes drafts and prereleases, so we get a stable channel for free.

## Test plan
- [x] `swift test` — all 349 tests pass (9 new in `UpdateCheckerTests`)
- [x] `./scripts/bundle.sh` — release archive builds and installs cleanly
- [ ] Launch app → About window shows current version status (live network check against `juliensimon/canopy` releases)
- [ ] Toggle "Check for updates on launch" off in Settings → relaunch issues no network request
- [ ] Click "Check Now" while offline → About shows "Couldn't check for updates" without crashing
- [ ] With `BuildInfo.version` temporarily set to `0.0.1`, launch and immediately switch apps → osascript notification appears

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: Canopy/Views/AboutView.swift [hunk 1]
```diff
@@ -81,6 +84,46 @@ struct AboutView: View {
...
+            case .available(let version, let url):
+                Text("New version \(version) available")
+                    .font(.caption)
+                    .foregroundStyle(.primary)
+                Button("Download") {
+                    NSWorkspace.shared.open(url)
+                }
+                .buttonStyle(.link)
+                .font(.caption)
+            case .failed:
```
[the remark this answers is not in the document]
Comment (juliensimon) [reply Canopy/Views/AboutView.swift:111]: Will bind the associated value (`case .failed(let message):`) and surface it in the About view's update status line.

File: Canopy.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -350,8 +356,7 @@
 			isa = PBXResourcesBuildPhase;
 			buildActionMask = 2147483647;
 			files = (
-				E403E5445BF739F5FBFBCA52 /* CanopyLogo.png in Resources */,
-				0037DE79842045E79C0B13A2 /* Assets.xcassets in Resources */,
+				D9227BD1898EF4FE74C4125D /* Assets.xcassets in Resources */,
 				56EC12B4E8F87D40411C840B /* CLAUDE.md in Resources */,
```
[the remark this answers is not in the document]
Comment (juliensimon) [reply Canopy.xcodeproj/project.pbxproj:360]: Will re-add `Resources/CanopyLogo.png` to the app target's Resources build phase so `Bundle.main.path(forResource:)` keeps working in Xcode builds.
